packages/kbn-i18n/src/core/i18n.test.js

```diff
@@ -0,0 +1,737 @@
+      const IntlRelativeFormat = require('intl-relativeformat');
+
+      i18n.setDefaultLocale('foo');
+
+      expect(IntlMessageFormat.defaultLocale).toBe('foo');
+      expect(IntlRelativeFormat.defaultLocale).toBe('foo');
+    });
+  });
+
+  describe('getDefaultLocale', () => {
+    test('should return "en" locale by default', () => {
+      expect(i18n.getDefaultLocale()).toBe('en');
+    });
+
+    test('should return updated locale', () => {
+      i18n.setDefaultLocale('foo');
+      expect(i18n.getDefaultLocale()).toBe('foo');
+    });
+  });
+
+  describe('setFormats', () => {
+    test('should throw error if formats parameter is not an object', () => {
+      expect(() => i18n.setFormats(undefined)).toThrow();
+      expect(() => i18n.setFormats(null)).toThrow();
+      expect(() => i18n.setFormats(true)).toThrow();
+      expect(() => i18n.setFormats(5)).toThrow();
+      expect(() => i18n.setFormats('foo')).toThrow();
+    });
+
+    test('should throw error if formats parameter is empty object', () => {
+      expect(() => i18n.setFormats({})).toThrow();
+    });
+
+    test('should merge current formats with a passed formats', () => {
+      expect(i18n.getFormats().date.short).not.toEqual({
+        month: 'short',
+        day: 'numeric',
+        year: 'numeric',
+      });
+
+      i18n.setFormats({
+        date: {
+          short: {
+            month: 'short',
+            day: 'numeric',
+            year: 'numeric',
+          },
+        },
+      });
+
+      expect(i18n.getFormats().date.short).toEqual({
+        month: 'short',
+        day: 'numeric',
+        year: 'numeric',
+      });
+
+      i18n.setFormats({
+        date: {
+          short: {
+            month: 'long',
+            day: 'numeric',
+            year: 'numeric',
+          },
+        },
+      });
+
+      expect(i18n.getFormats().date.short).toEqual({
+        month: 'long',
+        day: 'numeric',
+        year: 'numeric',
+      });
+    });
+  });
+
+  describe('getFormats', () => {
+    test('should return "en" formats by default', () => {
+      const { formats } = require('./formats');
+
+      expect(i18n.getFormats()).toEqual(formats);
+    });
+
+    test('should return updated formats', () => {
+      const { formats } = require('./formats');
+
+      i18n.setFormats({
+        foo: 'bar',
+      });
+
+      expect(i18n.getFormats()).toEqual({
+        ...formats,
+        foo: 'bar',
+      });
+    });
+  });
+
+  describe('getRegisteredLocales', () => {
+    test('should return empty array by default', () => {
+      expect(i18n.getRegisteredLocales()).toEqual([]);
+    });
+
+    test('should return array of registered locales', () => {
+      i18n.addMessages({
+        locale: 'en',
+      });
+
+      expect(i18n.getRegisteredLocales()).toEqual(['en']);
+
+      i18n.addMessages({
+        locale: 'ru',
+      });
+
+      expect(i18n.getRegisteredLocales()).toContain('en', 'ru');
```

Comment:
__nit:__ hmm, why not `toEqual(['en', 'ru'])` here instead of `toContain` + `toBe` (and below too)?

Reply:
`toContain` + `toBe` is more suitable here, as it does not take in account order of array.
Actually `toEqual` will work too, but I would not like to rely on it